### PR TITLE
Qi delete user backend

### LINKED ISSUE #84: Backend delete (hide) user
Add backend code to allow an admin to hide a user

### PR DESCRIPTION
**Pull request by publicqi**
## Overview

This PR adds code to allow an admin to delete (actually hide) a user. After hiding, the user does not have access to any api endpoints it previously had.

The added apis are endpoint "/api/admin/user/unhide" and "/api/admin/user/hide".
The implementation is done via adding a field to the User class.

Other changes are to adapt tests.

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 

## Linked Issues

Closes #84

### CODE REVIEW

**Review by iain801 — DISMISSED** [review]
Simple fixes lgtm

**Comment by dscpsyl** [comment]
The only extra thing I would bring up is that this has to work with the chat feature. So when a user is "deleted", we need to deal with their messages as well.

**Comment by publicqi** [comment]
> The only extra thing I would bring up is that this has to work with the chat feature. So when a user is "deleted", we need to deal with their messages as well.

I think there're two points worth noting: 
1. If you want to retrieve chat messages excluding hidden user's, you can use a function in chat Repository to filter
2. All endpoints should check if the requesting user is hidden. This is a general problem I mentioned in #86 Future Possibilities. Maybe there's a better way to decouple the code

Once the Chat backend is ready for merge, I can add the checks to that

**Comment by publicqi** [comment]
I'll fix the conflict when all other approved PRs get merged into main

File: src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java [hunk 1]
```diff
@@ -92,6 +98,14 @@ public void milkCows(JobContext ctx, Commons commons, UserCommons userCommons) {
...
      */
     public static double calculateMilkingProfit(Commons commons, UserCommons userCommons) {
+        // Because this is a private method, we trust that the
+        // caller has already checked that the user is not hidden
+
+        /*
+        if (userCommons.getUser().isHidden()) {
+            return 0.0;
+        }
+        */
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/jobs/MilkTheCowsJob.java:108]: Tell me more about this?   We still have commented out code here.  Are we not sure how to go about this?

**Comment by dscpsyl** [comment]
Close this after #106 has been merged

**Comment by pconrad** [comment]
Closing because has many merge conflicts, and the feature was not implemented in the way the user wanted.

**Pull request closed without merging**